File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/FnOr.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>Returns true if any one of the specified conditions evaluate to true, or returns false if all of the conditions evaluates to false.</summary>
    /// <remarks>
    /// <c>Fn::Or</c> acts as an OR operator. The minimum number
    /// of conditions that you can include is 2, and the maximum is 10.
    /// </remarks>
    [JsiiClass(nativeType: typeof(AlibabaCloud.SDK.ROS.CDK.Core.FnOr), fullyQualifiedName: "@alicloud/ros-cdk-core.FnOr", parametersJson: "[{\"docs\":{\"summary\":\"A condition that evaluates to true or false.\"},\"name\":\"condition\",\"type\":{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IRosConditionExpression\"}]}},\"variadic\":true}]")]
    public class FnOr : AlibabaCloud.SDK.ROS.CDK.Core.Intrinsic, AlibabaCloud.SDK.ROS.CDK.Core.IRosConditionExpression
    {
        /// <summary>Creates an ``Fn::Or`` condition function.</summary>
        /// <param name="condition">A condition that evaluates to true or false.</param>
        public FnOr(params object[] condition): base(_MakeDeputyProps(condition))
        {
        }

        [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
        private static DeputyProps _MakeDeputyProps(params object[] condition)
        {
            if (Amazon.JSII.Runtime.Configuration.RuntimeTypeChecking)
            {
                for (var __idx_596d40 = 0 ; __idx_596d40 < condition.Length ; __idx_596d40++)
                {
                    switch (condition[__idx_596d40])
                    {
                        case string cast_223119:
                            break;
                        case AlibabaCloud.SDK.ROS.CDK.Core.IRosConditionExpression cast_223119:
                            break;
                        case Amazon.JSII.Runtime.Deputy.AnonymousObject cast_223119:
                            // Not enough information to type-check...
                            break;
                        case null:
                            throw new System.ArgumentException($"Expected argument {nameof(condition)}[{__idx_596d40}] to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IRosConditionExpression).FullName}; received null", nameof(condition));
                        default:
                            throw new System.ArgumentException($"Expected argument {nameof(condition)}[{__idx_596d40}] to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IRosConditionExpression).FullName}; received {condition[__idx_596d40].GetType().FullName}", nameof(condition));
                    }
                }
            }
            return new DeputyProps(new object?[]{condition});
        }

        /// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
        /// <param name="reference">The Javascript-owned object reference</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected FnOr(ByRefValue reference): base(reference)
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from DeputyProps</summary>
        /// <param name="props">The deputy props</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected FnOr(DeputyProps props): base(props)
        {
        }
    }
}

File: multiple-languages/java/ros-cdk-core/src/main/java/com/aliyun/ros/cdk/core/FnOr.java
package com.aliyun.ros.cdk.core;

/**
 * Returns true if any one of the specified conditions evaluate to true, or returns false if all of the conditions evaluates to false.
 * <p>
 * <code>Fn::Or</code> acts as an OR operator. The minimum number
 * of conditions that you can include is 2, and the maximum is 10.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.85.0 (build 08ee592)", date = "2024-02-28T02:53:26.424Z")
@software.amazon.jsii.Jsii(module = com.aliyun.ros.cdk.core.$Module.class, fqn = "@alicloud/ros-cdk-core.FnOr")
public class FnOr extends com.aliyun.ros.cdk.core.Intrinsic implements com.aliyun.ros.cdk.core.IRosConditionExpression {

    protected FnOr(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected FnOr(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    /**
     * Creates an <code>Fn::Or</code> condition function.
     * <p>
     * @param condition A condition that evaluates to true or false. This parameter is required.
     */
    public FnOr(final @org.jetbrains.annotations.NotNull java.lang.Object... condition) {
        super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
        if (software.amazon.jsii.Configuration.getRuntimeTypeChecking()) {
            final java.util.List<java.lang.Object> __condition__asList = java.util.Arrays.asList(condition);
            for (int __idx_24648a = 0; __idx_24648a < __condition__asList.size(); __idx_24648a++) {
                final java.lang.Object __val_24648a = __condition__asList.get(__idx_24648a);
                if (
                     !(__val_24648a instanceof java.lang.String)
                    && !(__val_24648a instanceof com.aliyun.ros.cdk.core.IRosConditionExpression)
                    && !(__val_24648a.getClass().equals(software.amazon.jsii.JsiiObject.class))
                ) {
                    throw new IllegalArgumentException(
                        new java.lang.StringBuilder("Expected ")
                            .append("condition").append("[").append(__idx_24648a).append("]")
                            .append(" to be one of: java.lang.String, com.aliyun.ros.cdk.core.IRosConditionExpression; received ")
                            .append(__val_24648a.getClass()).toString());
                }
            }
        }
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this, java.util.Arrays.<Object>stream(condition).toArray(Object[]::new));
    }
}
